src/pet.js

```diff
@@ -0,0 +1,56 @@
+function Pet(name) {
+    this.name = name;
+    this.age = 0;
+    this.hunger = 0;
+    this.fitness = 10;
+    this.children = [];
+};
+
+Pet.prototype = {
+    get isAlive() {
+        return this.age < 30 && this.hunger < 10 && this.fitness > 0;
+    }
+  };
+
+Pet.prototype.growUp = function() {
+    if (!this.isAlive) throw new Error("Your pet is no longer alive :(");
+    this.age += 1;
+    this.hunger += 5;
+    this.fitness -= 3;
+};
+
+Pet.prototype.walk = function() {
+    const fitnessIncrementPerWalk = 4;
+    const maximumFitnessLevel = 10;
+    if (!this.isAlive) throw new Error("Your pet is no longer alive :(");
+    this.fitness <= 6 ? this.fitness += fitnessIncrementPerWalk : this.fitness += (maximumFitnessLevel - this.fitness);
```

Comment:
I would rewrite this ternary as follows `this.fitness <= maxmimFitnessLevel - fitnessIncrementPerWalk ? this.fitness += fitnessIncrementPerWalk : this.fitness = maximumFitnessLevel;`